test/PizzeriaTest.cpp

```diff
@@ -65,3 +65,33 @@ TEST_F(PizzeriaTest, calculatePriceForPizzaMock)
 
     delete mock;
 }
+
+TEST_F(PizzeriaTest, MainCppScenarioTest) {
+    // There is no need to expect_call for NiceMock 
+    StrictMock<PizzaMock> mockeritha{};
+    NiceMock<PizzaMock> mockeroni{};
+    
+    auto stubbioni = new PizzaStub{"Stubbioni"};
+
+    Pizzas pizzas = {&mockeritha, &mockeroni, stubbioni};
+
+    EXPECT_CALL(mockeritha, getPrice()).WillOnce(Return(22.50));
+    EXPECT_CALL(mockeroni, getPrice()).WillOnce(Return(25.00));
+    EXPECT_CALL(mockeritha, getName()).WillOnce(Return("Mockeritha"));
+    
+    auto orderId = pizzeria.makeOrder(pizzas);
+    auto calculatedPrice = pizzeria.calculatePrice(orderId);
+
+    EXPECT_CALL(mockeritha, getBakingTime()).WillOnce(Return(minutes(10)));
+
+    EXPECT_CALL(mt, sleep_for(minutes(0))).Times(1);
+    EXPECT_CALL(mt, sleep_for(minutes(10))).Times(1);
+    EXPECT_CALL(mt, sleep_for(minutes(1))).Times(1);
+
+    pizzeria.bakePizzas(orderId);
+    pizzeria.completeOrder(orderId);
+
+    ASSERT_EQ(calculatedPrice, 57.50);
```

Comment:
please avoid magic numbers. Use constexpr for values like 22.50, 25.00, minutes(10), etc.
Then use them in proper places. It shows dependencies between objects.